wallet/psbt.go

```diff
@@ -294,6 +309,52 @@ func addInputInfoSegWitV1(in *psbt.PInput, utxo *wire.TxOut,
 	}}
 }
 
+// addOutputInfo adds the BIP32 derivation info for an output from our internal
+// wallet.
+func addOutputInfo(out *psbt.POutput, txOut *wire.TxOut,
+	addr waddrmgr.ManagedPubKeyAddress) error {
+
+	keyScope, derivationPath, isKnown := addr.DerivationInfo()
+
+	// We don't know the derivation path for imported keys. Those shouldn't
+	// be selected as change outputs in the first place, but just to make
+	// sure we don't run into an issue, we return early for imported keys.
+	if !isKnown {
+		return fmt.Errorf("error adding output info to PSBT, change " +
+			"addr is an imported addr with unknown derivation " +
+			"path")
+	}
+
+	// Include the derivation path for each output in addition to the
+	// taproot specific info we have below.
+	derivation := &psbt.Bip32Derivation{
+		PubKey:               addr.PubKey().SerializeCompressed(),
+		MasterKeyFingerprint: derivationPath.MasterKeyFingerprint,
+		Bip32Path: []uint32{
+			keyScope.Purpose + hdkeychain.HardenedKeyStart,
+			keyScope.Coin + hdkeychain.HardenedKeyStart,
+			derivationPath.Account,
+			derivationPath.Branch,
+			derivationPath.Index,
+		},
+	}
+	out.Bip32Derivation = []*psbt.Bip32Derivation{
+		derivation,
+	}
+
+	// Include the Taproot derivation path as well if this is a P2TR output.
+	if txscript.IsPayToTaproot(txOut.PkScript) {
+		out.TaprootBip32Derivation = []*psbt.TaprootBip32Derivation{{
+			XOnlyPubKey:          derivation.PubKey[1:],
```

Comment:
Style nit: can capture this as `schnorrKey` above since it's used below also for `TaprootInternalKey`.